src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentTrainEnumerator.cs

```diff
@@ -0,0 +1,253 @@
+
+        public void WriteSpatial(int fieldId, string path, CoraxSpatialPointEntry entry)
+        {
+            // nothing to do here
+        }
+
+        public void Store(BlittableJsonReaderObject storedValue)
+        {
+            // nothing to do
+        }
+
+        public void RegisterEmptyOrNull(int fieldId, string fieldName, StoredFieldType type)
+        {
+            // nothing to do
+        }
+
+        public void Store(int fieldId, string name, BlittableJsonReaderObject storedValue)
+        {
+            // nothing to do
+        }
+
+        public void IncrementList()
+        {
+            
+        }
+
+        public int ResetList()
+        {
+            return default;
+        }
+
+        public void RestoreList(int old)
+        {
+        }
+
+        public void DecrementList()
+        {
+        }
+    }
+
+    private readonly DocumentsStorage _documentStorage;
+    private readonly QueryOperationContext _queryContext;
+    private readonly TransactionOperationContext _indexContext;
+    private readonly Index _index;
+    private readonly IndexType _indexType;
+    private readonly CoraxDocumentConverterBase _converter;
+    private readonly HashSet<string> _collections;
+    private readonly int _take;
+    private IEnumerator<ArraySegment<byte>> _itemsEnumerable;
+    private readonly List<(int FieldId, string FieldName, ByteString Value)> _terms;
+    private readonly Builder _builder;
+
+    public CoraxDocumentTrainEnumerator(TransactionOperationContext indexContext, CoraxDocumentConverterBase converter, Index index, IndexType indexType, DocumentsStorage storage, QueryOperationContext queryContext, HashSet<string> collections, int take = int.MaxValue)
+    {
+        _indexContext = indexContext;
+        _index = index;
+        _indexType = indexType;
+        _converter = converter;
+        _take = take;
+
+        _documentStorage = storage;
+        _queryContext = queryContext;
+        _collections = collections;
+        _terms = new List<(int FieldId, string FieldName, ByteString Value)>();
+        _builder = new Builder(indexContext.Allocator, _terms);
+    }
+
+    private IEnumerable<ArraySegment<byte>> GetItems()
+    {
+        var lowercaseAnalyzer = Analyzer.CreateLowercaseAnalyzer(_indexContext.Allocator);
+        var scope = new IndexingStatsScope(new IndexingRunStats());
+        
+        var wordsBuffer = new byte[1024];
+        var tokenBuffer = new Token[1024];
+        
+        foreach (var collection in _collections)
+        {
+            using var itemEnumerator = _index.GetMapEnumerator(GetItemsEnumerator(_queryContext, collection, _take), collection, _indexContext, scope, _indexType);
+            while (true)
+            {
+                if (itemEnumerator.MoveNext(_queryContext.Documents, out var mapResults, out var _) == false)
+                    break;
+
+                var doc = itemEnumerator.Current.Item as Document;
+                if (doc == null)
```

Comment:
when is this possible?

Reply:
Counters & Time Series,  which aren't relevent here for now

Reply:
Would be best to throw for now? or do a direct cast?

Reply:
done